Classes/ConnectionManager.php

```diff
@@ -254,6 +256,7 @@ public function getAllConnections()
                 $solrServer['solrPath'],
                 $solrServer['solrScheme']
             );
+            $this->checkAndSetAuthentication(end($connections), $solrServer);
```

Comment:
You could also assign the connection to a variable then pass it to you function and eventually push it onto `$connections`